sway-types/src/lib.rs

```diff
@@ -102,6 +102,8 @@ where
 pub struct ConfigTimeConstant {
     pub r#type: String,
     pub value: String,
+    #[serde(default)]
+    pub public: bool,
```

Comment:
While the `ContractId`s *currently* use the same method of being added to the namespace, they don't have anything to do with the concept of "configuration time" which will likely be removed in the future.

The `ConfigTimeConstant` is also implemented in a bit of a strange way which requires them to be parsed from a string. Instead, we likely want to construct the AST directly instead as mentioned [here](https://github.com/FuelLabs/sway/blob/8ca5234d85358a24e23434cbf4ef93c7674980c9/sway-core/src/semantic_analysis/namespace/module.rs#L84).

Could you open an issue to track this? Something along the lines of

> Contract ID constants should not use `ConfigTimeConstant` and should construct AST directly

Then we can leave this as-is to get this working for now, then come back and refactor this in a follow-up PR and I can help with the AST construction stuff.

Reply:
Created #3077 to track that 👍